attributes: Extending &default section, adding &default_insert (#201)
zeek/zeek-docs: script-reference/attributes.rst

    global mytable: table[count] of string &default="foo";

> Instead of constant values as shown above, the :zeek:attr:`&default` attribute
> accepts arbitrary Zeek expressions. For example, arithmetic expressions and

Reviewer: ```suggestion
In addition to constant values as shown above, the :zeek:attr:`&default` attribute
also accepts arbitrary Zeek expressions. For example, arithmetic expressions and
```

function calls are possible:

.. code-block:: zeek

   type Info: record {
       ts: time &log &default=network_time();
       ts_ms: double &log &default=time_to_double(network_time()) * 1000;
   };

The expressions are evaluated whenever a new record is instantiated.

> On tables, the :zeek:attr:`&default` attribute can further be set to a function
> or anonymously defined function (lambda). The signature of such a default function

Reviewer: ```suggestion
On tables, the :zeek:attr:`&default` attribute can further be set to a function
(including an anonymous lambda function), which will be invoked for any read access to a non-existing index to generate a substitute result. The signature of such a default function
```

> has to agree with the index and value types of the given table. Below, a default

Reviewer: ```suggestion
has to match the index and value types of the given table. Below, a default
```

function for a table with a composite index and value type of :zeek:type:`string` is shown.
The arguments for the function call, ``c`` and ``s`` below, are populated with
the values used for the index:

.. code-block:: zeek

    function table_default(c: count, s: string): string {
        return fmt("unknown-%s-%s", c, s);
    }

    global mytable: table[count, string] of string &default=table_default;

    print mytable[0, "a"];

> Using an anonymously defined function instead looks as follows:

Reviewer: ```suggestion
Using an anonymous function instead looks as follows:
```

.. code-block:: zeek

    global mytable: table[count, string] of string &default=function(c: count, s: string): string {
        return fmt("unknown-%s-%s", c, s);
    };

    print mytable[0, "a"];

> The output of both these examples is ``unknown-0-a``. Default functions are
> only invoked for non-existing indices.

Reviewer: ```suggestion
The output of both these examples is ``unknown-0-a``. 
```

        print a, b;
        }

.. zeek:attr:: &default_insert

&default_insert
---------------

.. versionadded:: 6.1

This attribute is only applicable to tables. :zeek:attr:`&default_insert`

> provides the same functionality as :zeek:attr:`&default` with the addition

Reviewer: ```suggestion
provides the same functionality as a table's :zeek:attr:`&default` but with the addition
```

> that upon access to a non-existing index, the created value will be inserted
> into the table. With :zeek:attr:`&default`, every table access for a missing
> index instead re-creates a new default value. Specifically for complex value
> types like tables or record types used for tracking further state,
> :zeek:attr:`&default_insert` is more useful than :zeek:attr:`&default`.

Reviewer: ```suggestion
that upon access to a non-existing index, the created value will be inserted
into the table. For complex value
types like tables or record types used for tracking further state,
:zeek:attr:`&default_insert` is often more useful and efficient than :zeek:attr:`&default`.
```
